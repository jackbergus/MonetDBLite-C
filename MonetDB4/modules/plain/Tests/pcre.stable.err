stderr of test 'pcre` in directory 'modules/plain` itself:


# 11:55:03 >  
# 11:55:03 >  Mtimeout -timeout 60 Mserver "--config=/ufs/zhang/install/monetdb/etc/MonetDB.conf" --debug=10 --set "monet_mod_path=/ufs/zhang/install/monetdb/lib/MonetDB:/ufs/zhang/install/monetdb/lib/bin" --set "gdk_dbfarm=/ufs/zhang/install/monetdb/var/MonetDB/dbfarm" --set "sql_logdir=/ufs/zhang/install/monetdb/var/MonetDB/log" --set mapi_port=32047 --set sql_port=46647 --set xquery_port=57257 --set monet_prompt= --trace --dbname=mTests_src_modules_plain  < pcre.milM
# 11:55:03 >  


# 11:55:03 >  
# 11:55:03 >  Done.
# 11:55:03 >  

