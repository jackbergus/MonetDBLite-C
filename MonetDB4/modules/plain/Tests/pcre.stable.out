stdout of test 'pcre` in directory 'modules/plain` itself:


# 11:55:03 >  
# 11:55:03 >  Mtimeout -timeout 60 Mserver "--config=/ufs/zhang/install/monetdb/etc/MonetDB.conf" --debug=10 --set "monet_mod_path=/ufs/zhang/install/monetdb/lib/MonetDB:/ufs/zhang/install/monetdb/lib/bin" --set "gdk_dbfarm=/ufs/zhang/install/monetdb/var/MonetDB/dbfarm" --set "sql_logdir=/ufs/zhang/install/monetdb/var/MonetDB/log" --set mapi_port=32047 --set sql_port=46647 --set xquery_port=57257 --set monet_prompt= --trace --dbname=mTests_src_modules_plain  < pcre.milM
# 11:55:03 >  

# Monet Database Server V4.9.3
# Copyright (c) 1993-2005, CWI. All rights reserved.
# Compiled for x86_64-redhat-linux-gnu/64bit with 64bit OIDs; dynamically linked.
# Visit http://monetdb.cwi.nl/ for further information.
module("pcre");

# Tests of replacing of a single string

pcre_replace("abracadabra", "bra", "*", "").print();
[ "a*cada*" ]
# result: "a*cada*"

pcre_replace("abracadabra", "a.*a", "*", "").print();
[ "*" ]
# result: "*"

pcre_replace("abracadabra", "a.*?a", "*", "").print();
[ "*c*bra" ]
# result: "*c*bra"

pcre_replace("abracadabra", "a", "", "").print();
[ "brcdbr" ]
# result: "brcdbr"

# Case _sensitive_ replacing
pcre_replace("abracadabraABRACADABRA", "a.*?a", "*", "").print();
[ "*c*braABRACADABRA" ]
# result: "*c*braABRACADABRA"

# Case _insensitive_ replacing
pcre_replace("abracadabraABRACADABRA", "a.*?a", "*", "i").print();
[ "*c*br*BR*D*" ]
# result: "*c*br*BR*D*"

# Multi-line string _without_ 'm' flag
pcre_replace("abcd\nabcd", "(^a)", "*", "").print();
[ "*bcd\nabcd" ]
# result: "*bcd\nabcd"

# Multi-line string _with_ 'm' flag
pcre_replace("abcd\nabcd", "(^a)", "*", "m").print();
[ "*bcd\n*bcd" ]
# result: "*bcd\n*bcd"

# Multi-line, case-insensitive
pcre_replace("abcd\nABCD", "(^a)", "*", "m").print();
[ "*bcd\nABCD" ]
# result: "*bcd\nABCD"

# Multi-line, case-sensitive
pcre_replace("abcd\nABCD", "(^a)", "*", "mi").print();
[ "*bcd\n*BCD" ]
# result: "*bcd\n*BCD"


# Tests of replacing of a BAT of strings
var mybat := bat(void, str);
mybat.append("aaaaabracadabra");
mybat.append("bbbbabracadabra");
mybat.append("ccccabracadabra");
mybat.append("ddddabracadabra");
mybat.append("aabbabracadabra");
mybat.append("abababracadabra");
mybat.append("aaccabracadabra");
mybat.append("acacabracadabra");
mybat.append("bcdbabracadabra");
mybat.append("abcdabracadabra");
mybat.append("ccddabracadabra");
mybat.append("cdcdabracadabra");
mybat.append("babracadabraddc");
mybat.append("abracadabra");
mybat.append("abracadabra\nABRACADABRA\nabracadabra\nABRACADABRA");
mybat:=mybat.reverse().mark(0@0).reverse();
#mybat.print();

# pcre_replace(mybat, "bra", "*", "")
pcre_replace(mybat, "bra", "*", "").print();
#---------------------------------------------------------#
# h	t		  # name
# void	str		  # type
#-------------------------#
[ 0@0,	  "aaaaa*cada*"	  ]
[ 1@0,	  "bbbba*cada*"	  ]
[ 2@0,	  "cccca*cada*"	  ]
[ 3@0,	  "dddda*cada*"	  ]
[ 4@0,	  "aabba*cada*"	  ]
[ 5@0,	  "ababa*cada*"	  ]
[ 6@0,	  "aacca*cada*"	  ]
[ 7@0,	  "acaca*cada*"	  ]
[ 8@0,	  "bcdba*cada*"	  ]
[ 9@0,	  "abcda*cada*"	  ]
[ 10@0,	  "ccdda*cada*"	  ]
[ 11@0,	  "cdcda*cada*"	  ]
[ 12@0,	  "ba*cada*ddc"	  ]
[ 13@0,	  "a*cada*"	  ]
[ 14@0,	  "a*cada*\nABRACADABRA\na*cada*\nABRACADABRA"	  ]

# pcre_replace(mybat, "a.*a", "*", "")
pcre_replace(mybat, "a.*a", "*", "").print();
#-------------------------------------------------#
# h	t		  # name
# void	str		  # type
#-------------------------#
[ 0@0,	  "*"		  ]
[ 1@0,	  "bbbb*"	  ]
[ 2@0,	  "cccc*"	  ]
[ 3@0,	  "dddd*"	  ]
[ 4@0,	  "*"		  ]
[ 5@0,	  "*"		  ]
[ 6@0,	  "*"		  ]
[ 7@0,	  "*"		  ]
[ 8@0,	  "bcdb*"	  ]
[ 9@0,	  "*"		  ]
[ 10@0,	  "ccdd*"	  ]
[ 11@0,	  "cdcd*"	  ]
[ 12@0,	  "b*ddc"	  ]
[ 13@0,	  "*"		  ]
[ 14@0,	  "*\nABRACADABRA\n*\nABRACADABRA"	  ]

# pcre_replace(mybat, "a.*?a", "*", "")
pcre_replace(mybat, "a.*?a", "*", "").print();
#---------------------------------------------------------#
# h	t		  # name
# void	str		  # type
#-------------------------#
[ 0@0,	  "***c*bra"	  ]
[ 1@0,	  "bbbb*c*bra"	  ]
[ 2@0,	  "cccc*c*bra"	  ]
[ 3@0,	  "dddd*c*bra"	  ]
[ 4@0,	  "*bb*c*bra"	  ]
[ 5@0,	  "*b*c*bra"	  ]
[ 6@0,	  "*cc*c*bra"	  ]
[ 7@0,	  "*c*c*bra"	  ]
[ 8@0,	  "bcdb*c*bra"	  ]
[ 9@0,	  "*br*d*"	  ]
[ 10@0,	  "ccdd*c*bra"	  ]
[ 11@0,	  "cdcd*c*bra"	  ]
[ 12@0,	  "b*c*braddc"	  ]
[ 13@0,	  "*c*bra"	  ]
[ 14@0,	  "*c*bra\nABRACADABRA\n*c*bra\nABRACADABRA"	  ]

# pcre_replace(mybat, "a", "", "")
pcre_replace(mybat, "a", "", "").print();
#---------------------------------------------------------#
# h	t		  # name
# void	str		  # type
#-------------------------#
[ 0@0,	  "brcdbr"	  ]
[ 1@0,	  "bbbbbrcdbr"	  ]
[ 2@0,	  "ccccbrcdbr"	  ]
[ 3@0,	  "ddddbrcdbr"	  ]
[ 4@0,	  "bbbrcdbr"	  ]
[ 5@0,	  "bbbrcdbr"	  ]
[ 6@0,	  "ccbrcdbr"	  ]
[ 7@0,	  "ccbrcdbr"	  ]
[ 8@0,	  "bcdbbrcdbr"	  ]
[ 9@0,	  "bcdbrcdbr"	  ]
[ 10@0,	  "ccddbrcdbr"	  ]
[ 11@0,	  "cdcdbrcdbr"	  ]
[ 12@0,	  "bbrcdbrddc"	  ]
[ 13@0,	  "brcdbr"	  ]
[ 14@0,	  "brcdbr\nABRACADABRA\nbrcdbr\nABRACADABRA"	  ]

# pcre_replace(mybat, "a.*?a", "*", "i")
pcre_replace(mybat, "a.*?a", "*", "i").print();
#-------------------------------------------------#
# h	t					  # name
# void	str					  # type
#-------------------------------------------------#
[ 0@0,	  "***c*bra"				  ]
[ 1@0,	  "bbbb*c*bra"				  ]
[ 2@0,	  "cccc*c*bra"				  ]
[ 3@0,	  "dddd*c*bra"				  ]
[ 4@0,	  "*bb*c*bra"				  ]
[ 5@0,	  "*b*c*bra"				  ]
[ 6@0,	  "*cc*c*bra"				  ]
[ 7@0,	  "*c*c*bra"				  ]
[ 8@0,	  "bcdb*c*bra"				  ]
[ 9@0,	  "*br*d*"				  ]
[ 10@0,	  "ccdd*c*bra"				  ]
[ 11@0,	  "cdcd*c*bra"				  ]
[ 12@0,	  "b*c*braddc"				  ]
[ 13@0,	  "*c*bra"				  ]
[ 14@0,	  "*c*bra\n*C*BRA\n*c*bra\n*C*BRA"	  ]

# pcre_replace(mybat, "(^a)", "*", "")
pcre_replace(mybat, "(^a)", "*", "").print();
#-----------------------------------------------------------------#
# h	t							  # name
# void	str							  # type
#-----------------------------------------------------------------#
[ 0@0,	  "*aaaabracadabra"					  ]
[ 1@0,	  "bbbbabracadabra"					  ]
[ 2@0,	  "ccccabracadabra"					  ]
[ 3@0,	  "ddddabracadabra"					  ]
[ 4@0,	  "*abbabracadabra"					  ]
[ 5@0,	  "*bababracadabra"					  ]
[ 6@0,	  "*accabracadabra"					  ]
[ 7@0,	  "*cacabracadabra"					  ]
[ 8@0,	  "bcdbabracadabra"					  ]
[ 9@0,	  "*bcdabracadabra"					  ]
[ 10@0,	  "ccddabracadabra"					  ]
[ 11@0,	  "cdcdabracadabra"					  ]
[ 12@0,	  "babracadabraddc"					  ]
[ 13@0,	  "*bracadabra"						  ]
[ 14@0,	  "*bracadabra\nABRACADABRA\nabracadabra\nABRACADABRA"	  ]

# pcre_replace(mybat, "(^a)", "*", "m")
pcre_replace(mybat, "(^a)", "*", "i").print();
#-----------------------------------------------------------------#
# h	t							  # name
# void	str							  # type
#-----------------------------------------------------------------#
[ 0@0,	  "*aaaabracadabra"					  ]
[ 1@0,	  "bbbbabracadabra"					  ]
[ 2@0,	  "ccccabracadabra"					  ]
[ 3@0,	  "ddddabracadabra"					  ]
[ 4@0,	  "*abbabracadabra"					  ]
[ 5@0,	  "*bababracadabra"					  ]
[ 6@0,	  "*accabracadabra"					  ]
[ 7@0,	  "*cacabracadabra"					  ]
[ 8@0,	  "bcdbabracadabra"					  ]
[ 9@0,	  "*bcdabracadabra"					  ]
[ 10@0,	  "ccddabracadabra"					  ]
[ 11@0,	  "cdcdabracadabra"					  ]
[ 12@0,	  "babracadabraddc"					  ]
[ 13@0,	  "*bracadabra"						  ]
[ 14@0,	  "*bracadabra\nABRACADABRA\nabracadabra\nABRACADABRA"	  ]

# pcre_replace(mybat, "(^a)", "*", "m")
pcre_replace(mybat, "(^a)", "*", "m").print();
#-----------------------------------------------------------------#
# h	t							  # name
# void	str							  # type
#-----------------------------------------------------------------#
[ 0@0,	  "*aaaabracadabra"					  ]
[ 1@0,	  "bbbbabracadabra"					  ]
[ 2@0,	  "ccccabracadabra"					  ]
[ 3@0,	  "ddddabracadabra"					  ]
[ 4@0,	  "*abbabracadabra"					  ]
[ 5@0,	  "*bababracadabra"					  ]
[ 6@0,	  "*accabracadabra"					  ]
[ 7@0,	  "*cacabracadabra"					  ]
[ 8@0,	  "bcdbabracadabra"					  ]
[ 9@0,	  "*bcdabracadabra"					  ]
[ 10@0,	  "ccddabracadabra"					  ]
[ 11@0,	  "cdcdabracadabra"					  ]
[ 12@0,	  "babracadabraddc"					  ]
[ 13@0,	  "*bracadabra"						  ]
[ 14@0,	  "*bracadabra\nABRACADABRA\n*bracadabra\nABRACADABRA"	  ]

# pcre_replace(mybat, "(^a)", "*", "mi")
pcre_replace(mybat, "(^a)", "*", "mi").print();
#-----------------------------------------------------------------#
# h	t							  # name
# void	str							  # type
#-----------------------------------------------------------------#
[ 0@0,	  "*aaaabracadabra"					  ]
[ 1@0,	  "bbbbabracadabra"					  ]
[ 2@0,	  "ccccabracadabra"					  ]
[ 3@0,	  "ddddabracadabra"					  ]
[ 4@0,	  "*abbabracadabra"					  ]
[ 5@0,	  "*bababracadabra"					  ]
[ 6@0,	  "*accabracadabra"					  ]
[ 7@0,	  "*cacabracadabra"					  ]
[ 8@0,	  "bcdbabracadabra"					  ]
[ 9@0,	  "*bcdabracadabra"					  ]
[ 10@0,	  "ccddabracadabra"					  ]
[ 11@0,	  "cdcdabracadabra"					  ]
[ 12@0,	  "babracadabraddc"					  ]
[ 13@0,	  "*bracadabra"						  ]
[ 14@0,	  "*bracadabra\n*BRACADABRA\n*bracadabra\n*BRACADABRA"	  ]

# Test if pcre_replace indeed only replace the matched string, and copy other
# strings unchanged
pcre_replace(mybat, "aaaa", "", "").print();
#-----------------------------------------------------------------#
# h	t			  # name
# void	str			  # type
#---------------------------------#
[ 0@0,	  "abracadabra"		  ]
[ 1@0,	  "bbbbabracadabra"	  ]
[ 2@0,	  "ccccabracadabra"	  ]
[ 3@0,	  "ddddabracadabra"	  ]
[ 4@0,	  "aabbabracadabra"	  ]
[ 5@0,	  "abababracadabra"	  ]
[ 6@0,	  "aaccabracadabra"	  ]
[ 7@0,	  "acacabracadabra"	  ]
[ 8@0,	  "bcdbabracadabra"	  ]
[ 9@0,	  "abcdabracadabra"	  ]
[ 10@0,	  "ccddabracadabra"	  ]
[ 11@0,	  "cdcdabracadabra"	  ]
[ 12@0,	  "babracadabraddc"	  ]
[ 13@0,	  "abracadabra"		  ]
[ 14@0,	  "abracadabra\nABRACADABRA\nabracadabra\nABRACADABRA"	  ]

# Test if the head column is kept after a replace operation
mybat := bat(int, str);
mybat.insert(int(20),  "aaaaabracadabra");
mybat.insert(int(1), "bbbbabracadabra");
mybat.insert(int(2), "ccccabracadabra");
mybat.insert(int(3), "ddddabracadabra");
mybat.insert(int(4), "aabbabracadabra");
mybat.insert(int(5), "abababracadabra");
mybat.insert(int(6), "aaccabracadabra");
mybat.insert(int(7), "acacabracadabra");
mybat.insert(int(8), "bcdbabracadabra");
mybat.insert(int(9), "abcdabracadabra");
mybat.insert(int(10), "ccddabracadabra");
mybat.insert(int(11), "cdcdabracadabra");
mybat.insert(int(12), "babracadabraddc");
mybat.insert(int(42), "abracadabra");
mybat.insert(int(30), "abracadabra\nABRACADABRA\nabracadabra\nABRACADABRA");
#mybat.print();

# pcre_replace(mybat, "(^a)", "*", "mi")
pcre_replace(mybat, "(^a)", "*", "mi").print();
#-----------------------------------------------------------------#
# h	t		  # name
# int	str		  # type
#-----------------------------------------------------------------#
[ 20,	  "*aaaabracadabra"					  ]
[ 1,	  "bbbbabracadabra"					  ]
[ 2,	  "ccccabracadabra"					  ]
[ 3,	  "ddddabracadabra"					  ]
[ 4,	  "*abbabracadabra"					  ]
[ 5,	  "*bababracadabra"					  ]
[ 6,	  "*accabracadabra"					  ]
[ 7,	  "*cacabracadabra"					  ]
[ 8,	  "bcdbabracadabra"					  ]
[ 9,	  "*bcdabracadabra"					  ]
[ 10,	  "ccddabracadabra"					  ]
[ 11,	  "cdcdabracadabra"					  ]
[ 12,	  "babracadabraddc"					  ]
[ 42,	  "*bracadabra"						  ]
[ 30,	  "*bracadabra\n*BRACADABRA\n*bracadabra\n*BRACADABRA"	  ]

# Test if pcre_replace indeed only replace the matched string, and copy other
# strings unchanged
# pcre_replace(mybat, "aaaa", "", "")
pcre_replace(mybat, "aaaa", "", "").print();
#-----------------------------------------------------------------#
# h	t			  # name
# int	str			  # type
#---------------------------------#
[ 20,	  "abracadabra"		  ]
[ 1,	  "bbbbabracadabra"	  ]
[ 2,	  "ccccabracadabra"	  ]
[ 3,	  "ddddabracadabra"	  ]
[ 4,	  "aabbabracadabra"	  ]
[ 5,	  "abababracadabra"	  ]
[ 6,	  "aaccabracadabra"	  ]
[ 7,	  "acacabracadabra"	  ]
[ 8,	  "bcdbabracadabra"	  ]
[ 9,	  "abcdabracadabra"	  ]
[ 10,	  "ccddabracadabra"	  ]
[ 11,	  "cdcdabracadabra"	  ]
[ 12,	  "babracadabraddc"	  ]
[ 42,	  "abracadabra"		  ]
[ 30,	  "abracadabra\nABRACADABRA\nabracadabra\nABRACADABRA"	  ]

# Test pcre_quote
var s := " !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_`abcdefghijklmnopqrstuvwxyz{|}~\t\n";
pcre_replace(s, pcre_quote(s), "matched", "").print();
[ "matched" ]

quit();

# 11:55:03 >  
# 11:55:03 >  Done.
# 11:55:03 >  

